components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java

```diff
@@ -0,0 +1,382 @@
+        this.access = this.mocker.getInstance(AuthorizationManager.class);
+        this.users = this.mocker.getInstance(UserManager.class);
+
+        this.userProfileDocument = new DocumentReference("wiki", "user", "00000001");
+        doReturn(this.currentUser).when(this.users).getCurrentUser();
+        doReturn(this.userProfileDocument).when(this.currentUser).getProfileDocument();
+
+        this.patientDocument = new DocumentReference("wiki", "data", "P0000001");
+        doReturn(this.patient).when(this.repository).getPatientById(this.id);
+        doReturn(this.patientDocument).when(this.patient).getDocument();
+
+        doReturn(new URI(this.uriString)).when(this.uriInfo).getRequestUri();
+        ReflectionUtils.setFieldValue(this.patientResource, "uriInfo", this.uriInfo);
+
+        Provider<XWikiContext> provider = this.mocker.getInstance(XWikiContext.TYPE_PROVIDER);
+        this.context = provider.get();
+    }
+
+    //----------------------------Get Patient Tests----------------------------
+
+    @Test
+    public void getPatientWhenRepositoryReturnsNullPatient()
+    {
+        doReturn(null).when(this.repository).getPatientById(anyString());
+
+        Response response = this.patientResource.getPatient(this.id);
+
+        verify(this.logger).debug("No such patient record: [{}]", this.id);
+        Assert.assertEquals(Status.NOT_FOUND.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void getPatientWhenUserDoesNotHaveAccess()
+    {
+        doReturn(false).when(this.access).hasAccess(Right.VIEW, this.userProfileDocument, this.patientDocument);
+
+        Response response = this.patientResource.getPatient(this.id);
+
+        verify(this.logger).debug("View access denied to user [{}] on patient record [{}]", this.currentUser, this.id);
+        Assert.assertEquals(Status.FORBIDDEN.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void checkGetPatientNormalBehaviour()
+    {
+        doReturn(true).when(this.access).hasAccess(Right.VIEW, this.userProfileDocument, this.patientDocument);
+        doReturn(new JSONObject()).when(this.patient).toJSON();
+
+        Response response = this.patientResource.getPatient(this.id);
+
+        Assert.assertTrue(response.getEntity() instanceof JSONObject);
+        Map<String, Map<String, String>> json = (Map<String, Map<String, String>>)response.getEntity();
+        Assert.assertThat(json, hasValue(hasEntry("rel", Relations.SELF)));
+        Assert.assertThat(json, hasValue(hasEntry("href", this.uriString)));
+        Map<String, List<MediaType>> actualMap = (Map)response.getMetadata();
+        Assert.assertThat(actualMap, hasValue(hasItem(MediaType.APPLICATION_JSON_TYPE)));
+        Assert.assertEquals(Status.OK.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void checkGetPatientAlwaysSendsLoggerMessageOnRequest()
```

Comment:
Logging is not an important part of the correct functionality of an implementation, so it doesn't have to be tested. Since it's not specified in the description of the interface, it can even be considered an implementation detail, which means that it shouldn't be tested at all.